src/tensor.cpp

```diff
@@ -458,6 +458,22 @@ std::size_t TensorDescriptor::GetNumBytes() const
 
 bool TensorDescriptor::IsPacked() const { return this->packed; }
 
+bool TensorDescriptor::IsContiguous() const
```

Comment:
Please add a comment that explains the physical meaning of this, i.e. what is an contiguous tensor. What is the difference from "packed"? And, if this check is Adam-specific, then please move this check to the adam solver.

Reply:
IsPacked() is only False when GetElementSize() != GetElementSpace(). 
Next example, tensor is packed and not contiguous.
```
>>> tensor = torch.tensor([[1,2,3,4],[5,6,7,8]])
>>> tensor = torch.as_strided(tensor, size=tensor.size(), stride=(1,2))
>>> tensor
tensor([[1, 3, 5, 7],
        [2, 4, 6, 8]])
>>> tensor.is_contiguous()
False
```
And this diff is taken from https://github.com/ROCm/MIOpen/pull/2970.
If https://github.com/ROCm/MIOpen/pull/2970 is merged first, it will be deleted from this PR.